sdk/keyvault/Azure.Security.KeyVault.Keys/src/KeyType.cs

```diff
@@ -21,4 +21,45 @@ public enum KeyType : uint
 
         All = uint.MaxValue
     }
+
+    public static class KeyTypeExtensions
+    {
+        public static KeyType ParseFromString(string value)
```

Comment:
Does case sensitivity matter here?

Reply:
Or in other enum extension types.

Reply:
I'm following what the server is expecting and the conventions are. i.e. for `KeyType` it is managed with UpperCase, but for `KeyOperations` it is lower case.
The `ParseFromString` takes care of it

Reply:
I think we should be case agnostic when parsing, but case specific when serializing

---
In reply to: [289959701](https://github.com/Azure/azure-sdk-for-net/pull/6457#discussion_r289959701) [](ancestors = 289959701)

Reply:
changed to follow that